osdk/tools/build_image.sh

```diff
@@ -8,8 +8,15 @@ SCRIPT_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
 ASTER_ROOT_DIR=${SCRIPT_DIR}/../..
 ASTER_RUST_VERSION=$( grep -m1 -o 'nightly-[0-9]\+-[0-9]\+-[0-9]\+' ${ASTER_ROOT_DIR}/rust-toolchain.toml )
 VERSION=$( cat ${ASTER_ROOT_DIR}/VERSION )
-IMAGE_NAME=asterinas/osdk:${VERSION}
-DOCKERFILE=${SCRIPT_DIR}/Dockerfile.ubuntu22.04
+DOCKERFILE=${SCRIPT_DIR}/ubuntu/Dockerfile
```

Comment:
Add a `docker` dir under `osdk/tools/` and move every file under the current `osdk/tools` into `osdk/tools/docker`. This makes it more apparent that all these files are for Docker, not for other tools.